### PR TITLE
Add externs for this.shadyRoot

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: polymer.externs.js [hunk 1]
```diff
@@ -41,6 +41,24 @@ PolymerElement.prototype.isAttached;
 PolymerElement.prototype.root;
 
 /**
+ * The shadow root for the element. 
+ * Only exists if running under Shadow Dom.
+ * You usually want to use `this.root`.
+ *
+ * @type {Node|undefined}
+ */
+PolymerElement.prototype.shadowRoot;
```
Comment (danbeam) [inline polymer.externs.js]: shouldn't this be http://www.w3.org/TR/shadow-dom/#the-shadowroot-interface ?

Comment (rictic) [inline polymer.externs.js]: Turns out that shadowRoot actually lives on HTMLElement, so this extern isn't necessary: https://github.com/google/closure-compiler/blob/master/externs/browser/html5.js#L1186

File: polymer.externs.js [hunk 2]
```diff
@@ -41,6 +41,15 @@ PolymerElement.prototype.isAttached;
 PolymerElement.prototype.root;
 
 /**
+ * The root node for the element.
+ * Only exists if running under Shady Dom.
+ * You usually want to use `this.root`.
+ *
+ * @type {Node|undefined}
```
Comment (jklein24) [inline polymer.externs.js]: Should this be {!Node|undefined} or {?Node|undefined}

Comment (rictic) [inline polymer.externs.js]: Updated to `{?Node|undefined}`

**Pull request merged**